hera_pspec/pspecdata.py

```diff
@@ -2171,17 +2199,32 @@ def pspec_run(dsets, filename, groupname=None, dset_labels=None, dset_pairs=None
         # get redundant baseline groups
         bls = None
 
+    # Construct dataset pairs to operate on
+    Ndsets = len(dsets)
+    if dset_pairs is None:
+        dset_pairs = list(itertools.combinations(range(Ndsets), 2))
+
+    if dset_labels is None:
+        dset_labels = ["dset{}".format(i) for i in range(Ndsets)]
+
     # load data if fed as filepaths
     if isinstance(dsets[0], (str, np.str)):
-        # load data into UVData objects if fed as list of strings
-        t0 = time.time()
-        _dsets = []
-        for d in dsets:
-            uvd = UVData()
-            uvd.read_miriad(d, bls=bls, polarizations=pols)
-            _dsets.append(uvd)
-        dsets = _dsets
-        utils.log("Loaded data in %1.1f sec." % (time.time() - t0), lvl=1, verbose=verbose)
+        try:
+            # load data into UVData objects if fed as list of strings
+            t0 = time.time()
+            _dsets = []
+            for i, dset in enumerate(dsets):
+                # read data
```

Comment:
Since this step is IO heavy, it might be very time-intensive, so should we print progress to screen?